Source/HelixToolkit.Wpf.SharpDX/Controls/MouseHandlers/ZoomHandler.cs

```diff
@@ -254,6 +254,20 @@ private void ChangeCameraDistance(double delta, Point3D zoomAround)
 
             var newTarget = zoomAround - newRelativeTarget;
             var newPosition = zoomAround - newRelativePosition;
+
+            var newDistance = (newPosition - zoomAround).Length;
+            var oldDistance = (this.Camera.Position - zoomAround).Length;
+
+            if (newDistance > this.Viewport.ZoomDistanceLimitFar && (oldDistance < this.Viewport.ZoomDistanceLimitFar || newDistance > oldDistance))
+            {
+                return;

```

Comment:
I think it would be better to adjust `newPosition` to the point with distance `ZoomDistanceLimitFar` from `the `zoomAround` point.